simplesat/sat/policy.py

```diff
@@ -13,7 +20,99 @@ def get_next_package_id(self, assignments, clauses):
         """
 
 
+class PolicyLogger(IPolicy):
```

Comment:
API-wise, I wonder if it would not be simpler to have the logger be a separate class (not implementing `IPolicy`), and create instances within policy classes.